Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/Settings.java

```diff
@@ -87,6 +90,55 @@ public boolean isGameSpecific()
     return !TextUtils.isEmpty(mGameId);
   }
 
+  public IniFile getWiimoteProfile(String profile, int padID)
+  {
+    File wiimoteProfileFile = SettingsFile.getWiiProfile(profile);
+    IniFile wiimoteProfileIni = new IniFile();
+
+    if (!mWiimoteProfileFiles.containsKey(wiimoteProfileFile))
+    {
+      wiimoteProfileIni.load(wiimoteProfileFile, false);
+
+      mWiimoteProfileFiles.put(wiimoteProfileFile, wiimoteProfileIni);
+    }
+
+    if (!mWiimoteProfileFiles.get(wiimoteProfileFile).exists(SECTION_PROFILE))
+    {
+      String defaultWiiProfilePath = DirectoryInitialization.getUserDirectory() +
+              "/Config/Profiles/Wiimote/WiimoteProfile.ini";
+
+      wiimoteProfileIni.load(defaultWiiProfilePath, false);
+
+      wiimoteProfileIni
+              .setString(SECTION_PROFILE, "Device", "Android/" + (padID + 4) + "/Touchscreen");
+
+      mWiimoteProfileFiles.put(wiimoteProfileFile, wiimoteProfileIni);
+    }
+
+    return mWiimoteProfileFiles.get(wiimoteProfileFile);
```

Comment:
With the above change, this could be something like this:

```java
if (!mWiimoteProfileFiles.containsKey(profile))
{
  IniFile newIni = new IniFile();
  newIni.load(SettingsFile.getWiiProfile(profile), false);
  mWiimoteProfileFiles.put(profile, wiimoteProfileIni);
}

IniFile wiimoteProfileIni = mWiimoteProfileFiles.get(profile);

if (!wiimoteProfileIni.exists(SECTION_PROFILE))
{
  String defaultWiiProfilePath = DirectoryInitialization.getUserDirectory() +
          "/Config/Profiles/Wiimote/WiimoteProfile.ini";

  wiimoteProfileIni.load(defaultWiiProfilePath, false);

  wiimoteProfileIni
          .setString(SECTION_PROFILE, "Device", "Android/" + (padID + 4) + "/Touchscreen");
}

return wiimoteProfileIni;
```

or (if Java 8 is available):

```java
IniFile wiimoteProfileIni = mWiimoteProfileFiles.computeIfAbsent(profile, profile -> {
  IniFile newIni = new IniFile();
  newIni.load(SettingsFile.getWiiProfile(profile), false);
  return newIni;
});

if (!wiimoteProfileIni.exists(SECTION_PROFILE))
{
  String defaultWiiProfilePath = DirectoryInitialization.getUserDirectory() +
          "/Config/Profiles/Wiimote/WiimoteProfile.ini";

  wiimoteProfileIni.load(defaultWiiProfilePath, false);

  wiimoteProfileIni
          .setString(SECTION_PROFILE, "Device", "Android/" + (padID + 4) + "/Touchscreen");
}

return wiimoteProfileIni;
```

Note that with this set-up, `wiimoteProfileIni` is mutated in place if `SECTION_PROFILE` does not exist, so it doesn't need to be put into the map again (we're mutating the one already in the map).